service/service.go

```diff
@@ -430,6 +474,9 @@ func (s *service) BeforeServe(
 		} else {
 			log.Println("No management servers found.")
 		}
+
+		opts.StorageArrays = make(map[string]StorageArrayConfig)
+		GetStorageArrays(secretParams, &opts)
```

Comment:
Not that you need to change this but given that GetStorageArrays is a function and not a method you should not pass in the entire structure if GetStorageArrays is only modifying one part of the structure. Passing in the address of StorageArrays is safer and you pass in only what the function needs. So you could have coded as:

```suggestion
		opts.StorageArrays = GetStorageArrays(secretParams)
```

In this way you simplify the code and reduce the amount of data that the function can modify.

Reply:
Don's comment makes sense. Please see if you update it to

opts.StorageArrays = GetStorageArrays(secretParams)

Reply:
will update in next PR.